src/xgov-dapp/src/features/rounds/index.tsx

```diff
@@ -22,51 +22,23 @@ const getRounds = (
   sortPredicate: Parameters<typeof sortBy>[1],
 ): VotingRoundGlobalState[] => {
   const filtered = rounds.filter(filterPredicate)
-  const sorted = sortBy(filtered, sortPredicate)
-  return sorted
+  return sortBy(filtered, sortPredicate)
 }
 
-const VotingRounds = () => {
+const VotingRoundsPage = () => {
   const { activeAddress } = useWallet()
   const creatorAddresses = useCreatorAddresses()
   const showMyRounds = creatorAddresses.length == 0 || creatorAddresses.includes('any')
   const isCreator = activeAddress && (creatorAddresses.includes(activeAddress) || creatorAddresses.includes('any'))
 
-  const [globalStates, setGlobalStates] = useState<VotingRoundGlobalState[]>([])
-  const [isLoading, setIsLoading] = useState(true)
-  const [error, setError] = useState<string | null>(null)
+  let addressesToFetch = [] as string[]
+  if (showMyRounds && activeAddress) {
+    addressesToFetch = [activeAddress]
+  } else if (!showMyRounds) {
+    addressesToFetch = creatorAddresses
+  }
 
-  useEffect(() => {
-    let addressesToFetch = [] as string[]
-    if (showMyRounds && activeAddress) {
-      addressesToFetch = [activeAddress]
-    } else if (!showMyRounds) {
-      addressesToFetch = creatorAddresses
-    }
-
-    if (addressesToFetch && addressesToFetch.length) {
-      ;(async () => {
-        setError(null)
-        setIsLoading(true)
-        try {
-          setGlobalStates(await fetchVotingRoundGlobalStatesByCreators(addressesToFetch))
-          setIsLoading(false)
-        } catch (e) {
-          setIsLoading(false)
-          if (e instanceof Error) {
-            setError(e.message)
-          } else {
-            // eslint-disable-next-line no-console
-            console.error(e)
-            setError('Unexpected error')
-          }
-        }
-      })()
-    } else {
-      setIsLoading(false)
-      setGlobalStates([])
-    }
-  }, [activeAddress, creatorAddresses, showMyRounds])
+  const { data: globalStates, isLoading, isError, error } = useGlobalStatesByCreatorQuery(addressesToFetch)
```

Comment:
would `isLoading` behave the same as before?

Reply:
Yes, `isLoading` is basically when there is no data